Tools/CLSGenerator.sh

```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
+
+# Source the helper functions
+source "$(dirname "$0")/Helper.sh"
+
+# Global variable definitions
+sourceDataPath="../Addons/Custom Lua Support/data/luafiles514/lua files"
+destinationDataPath="./Client/data/luafiles514/lua files"
+
+declare -a cls # Array to hold "[ Copied ]" status
+
+# Main CLSMenu loop
+CLSMenu() {
+    while true; do
+        clear
+        echo "==============="
+        echo "[1] All In One Package ${cls[1]}"
+        echo "[2] Headgears ${cls[2]}"
+        echo "[3] Random Options ${cls[3]}"
+        echo "[4] HatEffects ${cls[4]}"
+        echo "[5] NPC/Mob and Pets ${cls[5]}"
+        echo "[6] Lapine Boxes ${cls[6]}"
+        echo "[7] Signboards ${cls[7]}"
+        echo "[8] Robes/Costume Garments ${cls[8]}"
+        echo "[9] Weapons ${cls[9]}"
+        echo "[10] World Map ${cls[10]}"
+        echo "[11] Exit"
+        echo "==============="
+        read -p "Now choose: " type_choice
+
+        if ! [[ "$type_choice" =~ ^[0-9]+$ ]]; then
+            echo "Invalid input. Please enter a number."
+            pause_script
+            continue
+        fi
+
+        # Exit choice doesn't involve file operations
+        if [ "$type_choice" -eq 11 ]; then
+             echo "Exiting Custom Lua Support Generator."
+             exit 0
+        fi
+
+        # Remove existing cls directory if it exists, ignore errors
+        local cls_target_dir="./Client/data/luafiles514/lua files/cls"
+        if [ -d "$cls_target_dir" ]; then
+            echo "Removing existing directory: $cls_target_dir"
+            rm -rf "$cls_target_dir" 
+        fi
+        
+        local choice_successful=false # Flag to indicate a valid choice was processed
+        local copy_status=0 # 0 for success, 1 for failure
+        local item_failed=false # Track failure within multi-file choices
+
+        case "$type_choice" in
+            1) # All In One Package
+                echo "Copying All In One Package..."
+                # Replicating the broad copy from original batch
+                copy_dir_recursive "../Addons/Custom Lua Support/." "./Client/"; copy_status=$?
+                choice_successful=true
+                ;;
+            2) # Headgears
+                item_failed=false
+                copy_file "$sourceDataPath/cls/accessoryid.lub" "$destinationDataPath/cls/accessoryid.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/accname.lub" "$destinationDataPath/cls/accname.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/accname_f.lub" "$destinationDataPath/cls/accname_f.lub" || item_failed=true
```

Comment:
Remove this.

Reply:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL65